public/app/features/explore/state/query.ts

```diff
@@ -522,14 +527,14 @@ export const runQueries = createAsyncThunk<void, RunQueriesOptions>(
       absoluteRange,
       cache,
       supplementaryQueries,
-      correlationEditorHelperData,
     } = exploreItemState;
-    const isCorrelationEditorMode = getState().explore.correlationEditorDetails?.editorMode || false;
-    const isLeftPane = Object.keys(getState().explore.panes)[0] === exploreId;
-    const showCorrelationEditorLinks = isCorrelationEditorMode && isLeftPane;
-    const defaultCorrelationEditorDatasource = showCorrelationEditorLinks ? await getDataSourceSrv().get() : undefined;
-    const interpolateCorrelationHelperVars =
-      isCorrelationEditorMode && !isLeftPane && correlationEditorHelperData !== undefined;
+
+    const { defaultCorrelationEditorDatasource, scopedVars, showCorrelationEditorLinks } = await getCorrelationsData(
```

Comment:
@ifrost Since I made this change `public/app/features/explore/spec/interpolation.test.tsx` broke and I just can't figure out why.

Reply:
I now know why. The way the test was created required that runQueries() runs in sequence without adding an async function to the callback stack. Adding an await to a new function disrupted this expectation. I'm thinking test might be bit flaky, as it doesn't allow other parts of Explore to be refactored independently of this functionality.

After many many iterations, I was able to fix it by moving the correlations code before  `updateTime`.